Cargo.toml

```diff
@@ -59,6 +59,7 @@ bson = { git = "https://github.com/mongodb/bson-rust", branch = "main" }
 chrono = "0.4.7"
 derivative = "2.1.1"
 flate2 = { version = "1.0", optional = true }
+futures = "0.3.14"
```

Comment:
I think we don't need to explicitly pull in `futures-core`, `-util`, or `-executor` if we're using `futures`.

Reply:
We could do away with those but that would require changing every import of `futures-core`, `-util` and `-executor` in the driver. Is that something we want to do?

Reply:
The various individual `futures-x` crates have differing stability guarantees, which is why we depend on them directly. See [this comment](https://github.com/mongodb/mongo-rust-driver/pull/337#discussion_r627881130) for more context / reasoning.

Given that, I think we'll want to continue to depend on the various subcrates directly.

Reply:
Ah I see, that makes sense. Fixed!